gitlab-mailroom.yaml

```diff
@@ -0,0 +1,76 @@
+# source is gitlab so we can't use github updates to get expected commit
+# let's still auto create the PR, it will fail as expected commit will be wrong
+# however it will be easy to fix
+#nolint:git-checkout-must-use-github-updates
+package:
+  name: gitlab-mailroom
+  version: 0.0.23
+  epoch: 0
+  description: GitLab mail_room contains some merged functionality that GitLab requires, so this mirror fork is to help us release custom functionality.
+  copyright:
+    - license: MIT
+  dependencies:
+    runtime:
+      - jemalloc
+      - ruby-3.2
+      - busybox
+      - gitlab-cng-mailroom-scripts
+      - ruby3.2-webrick
+      - ruby3.2-bundler
+      - ruby3.2-rack-oauth2
+
+environment:
+  contents:
+    packages:
+      - build-base
+      - busybox
+      - icu-dev
+      - ruby-3.2
+      - ruby-3.2-dev
+      - ruby3.2-bundler
+      - zlib-dev
+
+vars:
+  gem: mail_room
+
+pipeline:
+  # GitLab-Mail_Room
+  - uses: git-checkout
+    with:
+      repository: https://gitlab.com/gitlab-org/ruby/gems/gitlab-mail_room.git
+      tag: v${{package.version}}
+      expected-commit: 578c314fc5d936978fafdea68952f9a5b2f7fe7d
+
+  - uses: ruby/unlock-spec
+
+  - runs: |
+      # This is your target directory for installing the gems
+      # I'm assuming it's an environment variable, change as needed
+      INSTALL_GEM_DIR=${{targets.destdir}}$(ruby -e 'puts Gem.default_dir')/
+
+      # GitLab-Mail_Room dependencies installed when running the package.
+      # These dependencies are registered as development dependencies in the repository
+      # but installed when using the binary.
+      gem install charlock_holmes:0.7.7 \
```

Comment:
This is a particular scenario here, they are installed in the Dockerfile https://github.com/chainguard-dev/image-requests/issues/455 but they are not marked as runtime deps in the gemfile. I am not sure what we want to take as the source of truth but considering this will end up in one of our images. I'd say we should install all the deps installed in the Dockerfile.

Reply:
Sorry, the link is this one https://gitlab.com/gitlab-org/build/CNG/-/blob/master/gitlab-mailroom/scripts/install-dependencies#L13.